modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc

```diff
@@ -87,80 +87,300 @@ $ docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig registr
 
 This allows your `kubeconfig` file to be consumed from inside the running container.
 
-[id="cnf-performing-end-to-end-tests-running-the-latency_tests"]
+[id="cnf-performing-end-to-end-tests-running-the-latency_tests_{context}"]
 === Running the latency tests
+
+[WARNING]
+====
+You must run the latency tests in Discovery mode. The latency tests can change the configuration of your cluster if you do not run in Discovery mode.
+====
+
 In {product-title} {product-version}, you can also run latency tests from the CNF-test container. The latency test allows you to set a latency limit so that you can determine performance, throughput, and latency.
 
-The latency test runs the `oslat` tool, which is an open source program to detect OS level latency. For more information, see the Red Hat Knowledgebase solution link:https://access.redhat.com/solutions/5315541[How to measure OS and hardware latency on isolated CPUs?].
+Three tools measure the latency of the system:
 
-By default, the latency tests are disabled. To enable the latency test, you must add the `LATENCY_TEST_RUN` variable and set its value to `true`. For example, `LATENCY_TEST_RUN=true`.
+* `hwlatdetect`
+* `cyclictest`
+* `oslat`
 
-Additionally, you can set the following environment variables for latency tests:
+Each tool has a specific use. Use the tools in sequence as the validation progresses.
 
-* `LATENCY_TEST_RUNTIME` - Specifies the amount of time (in seconds) that the latency test must run.
-* `OSLAT_MAXIMUM_LATENCY` - Specifies the maximum latency (in microseconds) that is expected from all buckets during the `oslat` test run.
+. The `hwlatdetect` tool measures the baseline that the bare hardware can achieve. Before proceeding with the next latency test, ensure that the number measured by `hwlatdetect` meets the required threshold because latency spikes can yield unreliable results.
```

Comment:
```suggestion
. The `hwlatdetect` tool measures the baseline that the bare hardware can achieve. Before proceeding with the next latency test, ensure that the number measured by `hwlatdetect` meets the required threshold because hardware latency spikes cannot be fixed by operating system tuning.
```